package-res/resources/web/pentaho/visual/base/View.js

```diff
@@ -205,24 +204,47 @@ define([
      * Decides how the visualization should react
      * to a modification of its properties.
      *
-     * Subclasses of `pentaho.visual.base.View` are expected to override this method
-     * with an implementation that
-     * selects the cheapest reaction to a generic change of properties.
-     * By default, this method invokes [_render]{@link pentaho.visual.base.View#_render},
-     * but the developer should also consider invoking
-     * [_resize]{@link pentaho.visual.base.View#_resize} or
-     * [_selectionChanged]{@link pentaho.visual.base.View#_selectionChanged}.
+     * By default, this method selects the cheapest reaction to a change of properties.
+     * It invokes:
+     * - [_resize]{@link pentaho.visual.base.View#_resize} when either of the properties
+     * [width]{{@link pentaho.visual.base.Model.Type#width} or
+     * [height]{{@link pentaho.visual.base.Model.Type#height} change,
+     * - [_selectionChanged]{@link pentaho.visual.base.View#_selectionChanged} when the property
+     * [selectionFilter]{{@link pentaho.visual.base.Model.Type#selectionFilter} changes
+     * - [_render]{@link pentaho.visual.base.View#_render} when any other property changes.
+     *
+     * Subclasses of `pentaho.visual.base.View` can override this method to
+     * extend the set of fast render methods.
      *
-     * @see pentaho.visual.base.View#_render
      * @see pentaho.visual.base.View#_resize
      * @see pentaho.visual.base.View#_selectionChanged
+     * @see pentaho.visual.base.View#_render
      *
-     * @param {Object} changeMap - Map of the properties that have changed.
+     * @param {!pentaho.lang.Changeset} changeset - Map of the properties that have changed.
      * @protected
      * @overridable
      */
-    _onChange: function(changeMap) {
-      this._render();
+    _onChange: function(changeset) {
+      var exclusionList = {
+        width: true,
+        height: true,
+        selectionMode: true, // never has a direct visual impact
+        selectionFilter: true
+      };
+      var fullUpdate = changeset.propertyNames.reduce(function(memo, prop){
+        return memo || !exclusionList[prop];
+      }, false);
+
+      if(fullUpdate) return this.render();
+
+      var updateSelection =  changeset.has("selectionFilter");
+      if(updateSelection){
```

Comment:
Yes it is.  It's unlikely to ever happen that "selectionFilter" and "height"/"width" change simultaneously, but it might be cheaper to do two cheap renders than a full render.
The price to pay is that a really stupid implementation would do two full renders.